rust/protocol/src/consts.rs

```diff
@@ -1,10 +1,34 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2022 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
-pub const MAX_FORWARD_JUMPS: usize = 25_000;
-pub const MAX_MESSAGE_KEYS: usize = 2000;
-pub const MAX_RECEIVER_CHAINS: usize = 5;
-pub const ARCHIVED_STATES_MAX_LENGTH: usize = 40;
-pub const MAX_SENDER_KEY_STATES: usize = 5;
+#![warn(missing_docs)]
+
+//! Magic numbers.
+
+/// Various positive integers bounding the maximum size of other data structures.
+pub mod limits {
+    /// The maximum number of encrypted messages that the client chain which decrypts Signal
+    /// messages in a [Double Ratchet] instance can retrieve at once (tracked in
+    /// [crate::proto::storage::session_structure::chain::ChainKey::index] as well as a separate
+    /// `counter`).
```

Comment:
This isn't really about how many messages you can decrypt at once; it's about how many messages you can *not* receive and still decrypt further messages on an existing chain. Hard to explain, though!

Reply:
I have spent some time to distill what it actually does, which ended up taking a couple paragraphs. Please let me know if I should move that documentation elsewhere.

Reply:
(this is now in `limits.rs`)